src/backend/renderer/mod.rs

```diff
@@ -289,7 +341,9 @@ pub fn buffer_dimensions(
 }
 
 /// Returns the dimensions of a wl_buffer
-#[cfg(all(feature = "wayland_frontend", not(feature = "backend_egl")))]
+///
+/// *Note*: This will only return dimensions for buffer types known to smithay (see [`buffer_type`])
+#[cfg(all(feature = "wayland_frontend", not(feature = "backend_egl"), not(feature = "use_system_lib")))]
```

Comment:
Same for this function, it is undefined if we have `use_system_lib` but not `backend_egl`.